receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -130,18 +131,16 @@ func (tsm *timeseriesMap) get(metric pmetric.Metric, kv pcommon.Map) (*timeserie
 }
 
 // Create a unique string signature for attributes values sorted by attribute keys.
-// NOTE: this function mutates the attributes map as a side effect.
-func getAttributesSignature(kv pcommon.Map) string {
-	labelValues := make([]string, 0, kv.Len())
-	kv.Sort()
-	kv.Range(func(_ string, attrValue pcommon.Value) bool {
+func getAttributesSignature(m pcommon.Map) [16]byte {
+	clearedMap := pcommon.NewMap()
+	m.Range(func(k string, attrValue pcommon.Value) bool {
 		value := attrValue.Str()
 		if value != "" {
```

Comment:
We need to investigate why is this condition here, without this the hash implementation will be an improvement. If an empty label is equivalent with missing label (which seem to be the case here), then we can ensure that when we convert from prom data we don't add the empty labels, so here will not have to create a new map.

Reply:
Sounds good to me. Created an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17901